src/OnboardingSPA/pages/Steps/SitePages/index.js

```diff
@@ -81,28 +83,52 @@ const StepSitePages = () => {
 						.map( ( checkedPage ) => {
 							return checkedPage.slug;
 						} );
-					handleCheckedPages( checkedPages );
+					handleCheckedPages( checkedPages, sitePagesResponse.body );
 				}
 			}
 		}
 		setIsLoaded( true );
 	};
 
-	const handleCheckedPages = ( selectedPages ) => {
+	const stateToFlowData = ( selectedPages, sitePages ) => {
+		return sitePages !== false
+			? sitePages
+					?.filter( ( sitePage ) => {
```

Comment:
Filter and Map are gonna first iterate the array once then the selected list is iterated again i.e. 4 + 3 for the site pages step, maybe you can try something like reduce? (This would do that in 4)

```suggestion
					var checkedPagesCopy = sitePagesResponse.body.reduce(function (result, sitePage) {
						if (sitePage.selected)
							result.push( sitePage.slug );
						return result;
					}, []);
```

https://stackoverflow.com/questions/34398279/map-and-filter-an-array-at-the-same-time#:~:text=You%20should%20use%20Array.reduce%20for%20this.